src/main/java/edu/cornell/kfs/fp/service/impl/AmazonWebServicesBillingServiceImpl.java

```diff
@@ -0,0 +1,446 @@
+                LOG.info("processAmazonAccountDetails() Found an internal kfs account, ignoring this transaction for AWS account " 
+                    + amazonAccountDetail.getAwsAccount());
+            } else {
+                Account account = determineWhichAccountToUse(amazonAccountDetail.getCostCenter(), amazonAccountDetail.getKfsAccount());
+                if (account != null) {
+                    KualiDecimal cost = new KualiDecimal(Double.valueOf(amazonAccountDetail.getCost()));
+                    if (shouldBuildDistributionOfIncomeDocument(amazonAccountDetail, cost)) {
+                        try {
+                            buidAndRouteDistributionOfIncomeDocument(account, cost, amazonAccountDetail);
+                        } catch (Throwable e) {
+                            valid = false;
+                            LOG.error("processAmazonAccountDetails() there waw a problem building a DI for " +
+                                amazonAccountDetail.toString(), e);
+                            GlobalVariables.getMessageMap().getErrorMessages().clear();
+                        }
+                    }
+                }
+            }
+            count ++;
+        }
+        LOG.info("Processed " + count + " AWS records");
+        return valid;
+    }
+    
+    private Account determineWhichAccountToUse(String costCenter, String kfsAccountString) {
+        Account costAccount = buildAccount(costCenter);
+        if (validAccount(costAccount)) {
+            return costAccount;
+        }
+        
+        Account kfsAccount = buildAccount(kfsAccountString);
+        if (validAccount(kfsAccount)) {
+            return kfsAccount;
+        }
+        LOG.error("determineWhichAccountToUse() unable to find an active account, both cost account " + costCenter + " and kfs account " + kfsAccountString + " are invalid");
+        return null;
+    }
+    
+    private Account buildAccount(String accountNumber) {
+        if (StringUtils.isNotBlank(accountNumber)) {
+            try {
+                Account account = getAccountService().getUniqueAccountForAccountNumber(accountNumber);
+                return account;
+            } catch (Exception e) {
+                LOG.debug(accountNumber + " is an invalid account");
+            }
+        }
+        return null;
+    }
+    
+    private boolean validAccount(Account account) {
+        return account != null && account.isActive() && !account.isExpired();
+    }
+    
+    private boolean shouldBuildDistributionOfIncomeDocument(AmazonAccountDetail amazonAccountDetail, KualiDecimal cost) {
+        for (DocumentHeader dh : findDocumentHeadersForAmazonDetail(amazonAccountDetail)) {
+            try {
+                DistributionOfIncomeAndExpenseDocument di = (DistributionOfIncomeAndExpenseDocument) getDocumentService().getByDocumentHeaderId(dh.getDocumentNumber());
+                if (di.getTotalDollarAmount().equals(cost)) {
```

Comment:
This is only checking the cost, do you need to check the period as well? What if the cost is the same two months in a row?